src/overlay016/battle_controller.c

```diff
@@ -4035,279 +4036,251 @@ static void ov16_02250798 (BattleSystem * param0, BattleContext * param1)
+                        monsAlive++;
+                        if (battleCtx->selectedPartySlot[i ^ 2] != j) {
+                            totalHP += curHP;
                         }
                     }
+                }
 
-                    if (v7 == 0) {
-                        param1->battlersSwitchingMask |= FlagIndex(v1);
-                        param1->selectedPartySlot[v1] = 6;
-                    } else {
-                        param1->commandNext = v4;
-                        param1->command = 21;
-                        param1->battlerStatusFlags[v1] |= 0x1;
-                    }
+                if (totalHP == 0) {
+                    battleCtx->battlersSwitchingMask |= FlagIndex(i);
+                    battleCtx->selectedPartySlot[i] = 6;
+                } else {
+                    battleCtx->commandNext = retCommand;
+                    battleCtx->command = BATTLE_CONTROL_EXEC_SCRIPT;
+                    battleCtx->battlerStatusFlags[i] |= BATTLER_STATUS_SWITCHING;
                 }
             }
-        } else {
-            if (param1->battleMons[v1].curHP == 0) {
-                {
-                    int v12;
-                    int v13 = 0;
-                    Party * v14;
-                    Pokemon * v15;
-                    UnkStruct_ov16_0225BFFC * v16;
-
-                    v14 = ov16_0223DF20(param0, v1);
-                    v16 = BattleSystem_BattlerData(param0, v1);
-
-                    for (v12 = 0; v12 < Party_GetCurrentCount(v14); v12++) {
-                        v15 = Party_GetPokemonBySlotIndex(v14, v12);
-
-                        if ((Pokemon_GetValue(v15, MON_DATA_SPECIES_EGG, NULL) != 0) && (Pokemon_GetValue(v15, MON_DATA_SPECIES_EGG, NULL) != 494)) {
-                            v13 += Pokemon_GetValue(v15, MON_DATA_CURRENT_HP, NULL);
-                        }
-                    }
+        } else if (battleCtx->battleMons[i].curHP == 0) {
+            int j;
+            int curHP = 0;
+            Party *party = BattleSystem_Party(battleSys, i);
+            BattlerData *battlerData = BattleSystem_BattlerData(battleSys, i);
 
-                    if (v13 == 0) {
-                        param1->battlersSwitchingMask |= FlagIndex(v1);
-                        param1->selectedPartySlot[v1] = 6;
-                    } else {
-                        param1->commandNext = v4;
-                        param1->command = 21;
-                        param1->battlerStatusFlags[v1] |= 0x1;
-                    }
+            for (j = 0; j < Party_GetCurrentCount(party); j++) {
+                Pokemon *pokemon = Party_GetPokemonBySlotIndex(party, j);
+
+                if (Pokemon_GetValue(pokemon, MON_DATA_SPECIES_EGG, NULL) != FALSE
+                        && Pokemon_GetValue(pokemon, MON_DATA_SPECIES_EGG, NULL) != SPECIES_EGG) {
+                    curHP += Pokemon_GetValue(pokemon, MON_DATA_CURRENT_HP, NULL);
```

Comment:
I tried to figure out an inline for this general procedure (since it's used in a bunch of places), but wasn't able to quite nail it down. Leaving it as-is.